### ISSUE TITLE
Ability to use SQLite without native dependency

### ISSUE THREAD

**Issue by davidindra** [dotnet/efcore#16910]
Hello,
I am interested in some way to use Entity Framework Core with SQLite DB without the need of using native DLL dependency. For my internal usage, I've done some modifications to depend on this https://github.com/CsharpDatabase/CsharpSQLite repo instead. Would it be possible to include these modifications as part of/as an alternative in official EF Core repo? Or is there another way, how to use SQLite without native DLL dependency? :)

Have a nice day!
D.

**Comment by ErikEJ** [comment]
@davidindra That port is 6 years old, and will be completely lagging behind the current version of SQLite.

**Comment by davidindra** [comment]
@ErikEJ That's true, but it works and the advantage it would make is big I think.

**Comment by ajcvickers** [comment]
@davidindra We would certainly be supportive of an open-source, community driven provider for EF Core using some form of managed SQLite. However, it's not something we're planning to take into this repo and manage--we don't scale to all the possible providers out there--which is why a community driven project is preferable here.

**Comment by ericsink** [comment]
FWIW, this is something I have been investigating, but I don't have anything ready yet.

See this blog entry for a summary of one approach I have explored:

https://ericsink.com/entries/wasm_wasi_dotnet.html

I have also made some progress with an alternate approach, wherein I compile LLVM bitcode directly to CIL.  I have not (yet) open sourced this effort, and it is not as far along as the wasm-based stuff mentioned above, but it would bring certain advantages.  For example, one problem with the wasm approach is that it would require working around wasm's sandbox.

I hope to eventually end up with a toolchain that can compile SQLite directly to CIL, and then to include such a build as one of the offerings for SQLitePCLRaw.
